docs/source/transaction_family_specifications/identity_transaction_family.rst

```diff
@@ -0,0 +1,217 @@
+
+
+State
+=====
+
+Policies
+--------
+A policy will have a name and a list of entries. Each policy entry will have a
+type and a key list. The type will be either PERMIT_KEY or DENY_KEY and the key
+list will be a list of public keys.
+
+.. code-block:: protobuf
+
+  // Policy will be stored in a PolicyList to account for hash collisions
+  message PolicyList {
+    repeated Policy policies = 1;
+  }
+
+  message Policy {
+    enum Type {
+      PERMIT_KEY = 0;
+      DENY_KEY = 1;
+    }
+
+    message Entry {
+      // Whether this is a PERMIT_KEY or DENY_KEY entry
+      Type type = 1;
+      // This should be a public key or * to refer to all participants.
+      string key = 2;
+
+    }
+
+    // name of the policy, this should be unique.
+    string name = 1;
+
+    // The entries will be processed in order from first to last.
+    repeated Entry entries = 2;
+  }
+
+Roles
+-----
+A role will be made up of a role name and the name of the policy to be enforced
+for that role. The data will be stored in state at the address described above
+using the following protobuf messages:
+
+.. code-block:: protobuf
+
+  // Roles will be stored in a RoleList to handle state collisions. The Roles
+  // will be sorted by name.
+  message RoleList {
+    repeated Role roles = 1;
+  }
+
+  message Role{
+    // Role name
+    string name = 1;
+
+    // Name of corresponding policy
+    string policy_name = 2;
+  }
+
+Addressing
+----------
+All identity data will be stored under the special namespace of “00001d”.
+
+For each policy, the address will be formed by concatenating the namespace, the
+special policy namespace of “00”, and the first 62 characters of the SHA-256
+hash of the policy name:
+
+.. code-block:: pycon
+
+ >>> "00001d" + "00" + hashlib.sha256(policy_name.encode()).hexdigest()[:62]
+
+Address construction for roles will follow a pattern similar to address
+construction in the settings namespace. Role names will be broken into four
+parts, where parts of the string are delimited by the "." character. For
+example, the key a.b.c would be split into the parts "a", "b", "c", and the
+empty string. If a key would have more than four parts the extra parts are left
+in the last part. For example, the key a.b.c.d.e would be split into "a", "b",
+"c", and "d.e".
+
+A short hash is computed for each part. For the first part the first 14
+characters of the SHA-256 hash are used. For the remaining parts the first 16
+characters of the SHA-256 hash are used. The address is formed by concatenating
+the identity namespace “00001d”, the role namespace “01”, and the four short
+hashes.
+
+For example, the address for the role client.query_state would be constructed
+as follows:
+
+.. code-block:: pycon
+
+  >>> "00001d"+ "01" + hashlib.sha256('client'.encode()).hexdigest()[:14]+ \
+    hashlib.sha256('query_state'encode()).hexdigest()[:16]+ \
+    hashlib.sha256(''encode()).hexdigest()[:16]+ \
+    hashlib.sha256(''encode()).hexdigest()[:16]
```

Comment:
Missing a dot before the `encode` on the last three here.

i.e. `'query_state'encode()` -> `'query_state'.encode()`